src/util/messagepipe.h

```diff
@@ -71,26 +71,27 @@ class TwoWayMessagePipe {
     // the first is the sender's pipe (sends SenderMessageType and receives
     // ReceiverMessageType messages) and the second is the receiver's pipe
     // (sends ReceiverMessageType and receives SenderMessageType messages).
-    static QPair<MessagePipe<SenderMessageType, ReceiverMessageType>*,
-                 MessagePipe<ReceiverMessageType, SenderMessageType>*> makeTwoWayMessagePipe(
-                     int sender_fifo_size,
-                     int receiver_fifo_size) {
+    static std::pair<
+            std::unique_ptr<MessagePipe<SenderMessageType, ReceiverMessageType>>,
+            std::unique_ptr<MessagePipe<ReceiverMessageType, SenderMessageType>>>
+    makeTwoWayMessagePipe(
+            int sender_fifo_size,
+            int receiver_fifo_size) {
         QSharedPointer<TwoWayMessagePipe<SenderMessageType, ReceiverMessageType>> pipe(
                 new TwoWayMessagePipe<SenderMessageType, ReceiverMessageType>(
                         sender_fifo_size, receiver_fifo_size));
 
-        return QPair<MessagePipe<SenderMessageType, ReceiverMessageType>*,
-                MessagePipe<ReceiverMessageType, SenderMessageType>*>(
-                new MessagePipe<SenderMessageType, ReceiverMessageType>(
+        return {
+                std::move(std::make_unique<MessagePipe<SenderMessageType, ReceiverMessageType>>(
```

Comment:
Pretty sure you can omit the `std::move`s here.

Reply:
yes, it works.